main.tf

```diff
@@ -32,6 +32,921 @@ locals {
+  atomic                     = try(var.sagemaker.atomic, null)
+  skip_crds                  = try(var.sagemaker.skip_crds, null)
+  render_subchart_notes      = try(var.sagemaker.render_subchart_notes, null)
+  disable_openapi_validation = try(var.sagemaker.disable_openapi_validation, null)
+  wait                       = try(var.sagemaker.wait, false)
+  wait_for_jobs              = try(var.sagemaker.wait_for_jobs, null)
+  dependency_update          = try(var.sagemaker.dependency_update, null)
+  replace                    = try(var.sagemaker.replace, null)
+  lint                       = try(var.sagemaker.lint, null)
+
+  postrender = try(var.sagemaker.postrender, [])
+
+  set = concat([
+    {
+      # shortens pod name from `ack-sagemaker-sagemaker-chart-xxxxxxxxxxxxx` to `ack-sagemaker-xxxxxxxxxxxxx`
+      name  = "nameOverride"
+      value = "ack-sagemaker"
+    },
+    {
+      name  = "aws.region"
+      value = local.region
+    },
+    {
+      name  = "serviceAccount.name"
+      value = local.sagemaker_name
+    }],
+    try(var.sagemaker.set, [])
+  )
+  set_sensitive = try(var.sagemaker.set_sensitive, [])
+
+  # IAM role for service account (IRSA)
+  set_irsa_names                = ["serviceAccount.annotations.eks\\.amazonaws\\.com/role-arn"]
+  create_role                   = try(var.sagemaker.create_role, true)
+  role_name                     = try(var.sagemaker.role_name, "ack-sagemaker")
+  role_name_use_prefix          = try(var.sagemaker.role_name_use_prefix, true)
+  role_path                     = try(var.sagemaker.role_path, "/")
+  role_permissions_boundary_arn = lookup(var.sagemaker, "role_permissions_boundary_arn", null)
+  role_description              = try(var.sagemaker.role_description, "IRSA for Sagemaker controller for ACK")
+  role_policies = lookup(var.sagemaker, "role_policies", {
+    core_policy                      = var.enable_sagemaker ? aws_iam_policy.sagemaker_core_policy[0].arn : null,
+    studio_policy                    = var.enable_sagemaker ? aws_iam_policy.sagemaker_studio_policy[0].arn : null,
+    space_management_policy          = var.enable_sagemaker ? aws_iam_policy.sagemaker_space_management_policy[0].arn : null,
+    aws_service_actions_policy       = var.enable_sagemaker ? aws_iam_policy.aws_service_actions_policy[0].arn : null,
+    resource_specific_actions_policy = var.enable_sagemaker ? aws_iam_policy.resource_specific_actions_policy[0].arn : null,
+    s3_actions_policy                = var.enable_sagemaker ? aws_iam_policy.s3_actions_policy[0].arn : null
+  })
+
+  create_policy = try(var.sagemaker.create_policy, false)
+
+  oidc_providers = {
+    this = {
+      provider_arn = local.oidc_provider_arn
+      # namespace is inherited from chart
+      service_account = local.sagemaker_name
+    }
+  }
+
+  tags = var.tags
+}
+
+# recommended sagemaker-controller policy https://github.com/aws-controllers-k8s/sagemaker-controller/blob/main/config/iam/recommended-policy-arn
```

Comment:
this is different than the recommended?
The recommended is just `arn:aws:iam::aws:policy/AmazonSageMakerFullAccess`